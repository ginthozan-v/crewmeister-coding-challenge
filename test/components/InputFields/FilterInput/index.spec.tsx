import React from 'react';
import { render } from '@testing-library/react';
import Filter from '../../../../components/InputFields/FilterInput';

describe('Filter Component', () => {
  test('should match the snapshot', () => {
    const { container } = render(
      <Filter
        label="Filter by absence type"
        data={[
          { label: 'sickness', value: 'sickness' },
          { label: 'vacation', value: 'vacation' },
        ]}
        setVacationType={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });

  test('should check the options', () => {
    const view = render(
      <Filter
        label="Filter by absence type"
        data={[
          { label: 'sickness', value: 'sickness' },
          { label: 'vacation', value: 'vacation' },
        ]}
        setVacationType={() => {}}
      />
    );
    expect(view.container).toMatchSnapshot();
  });
});
