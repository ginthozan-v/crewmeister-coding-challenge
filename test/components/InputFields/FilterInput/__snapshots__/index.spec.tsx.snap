// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component should check the options 1`] = `
<div>
  <p
    class="title"
  >
    Filter by absence type
  </p>
  <select
    class="input"
    id="absenceType"
    name="absenceType"
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="sickness"
    >
      sickness
    </option>
    <option
      value="vacation"
    >
      vacation
    </option>
  </select>
</div>
`;

exports[`Filter Component should match the snapshot 1`] = `
<div>
  <p
    class="title"
  >
    Filter by absence type
  </p>
  <select
    class="input"
    id="absenceType"
    name="absenceType"
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="sickness"
    >
      sickness
    </option>
    <option
      value="vacation"
    >
      vacation
    </option>
  </select>
</div>
`;

exports[`I want to filter absences by type should check the options 1`] = `
<div>
  <p
    class="title"
  >
    Filter by absence type
  </p>
  <select
    class="input"
    id="absenceType"
    name="absenceType"
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="sickness"
    >
      sickness
    </option>
    <option
      value="vacation"
    >
      vacation
    </option>
  </select>
</div>
`;

exports[`I want to filter absences by type should match 1`] = `
<div>
  <p
    class="title"
  >
    Filter by absence type
  </p>
  <select
    class="input"
    id="absenceType"
    name="absenceType"
  >
    <option
      value=""
    >
      All
    </option>
    <option
      value="sickness"
    >
      sickness
    </option>
    <option
      value="vacation"
    >
      vacation
    </option>
  </select>
</div>
`;
